content/developers/dogstatsd/data_types.md

```diff
@@ -0,0 +1,309 @@
+
+For Python:
+```python
+
+def upload_file(file):
+    statsd.increment('file_service.bytes_uploaded', file.size())
+    save_file(file)
+    return 'File uploaded!'
+```
+
+Note that for counters coming from another source that are ever-increasing and never reset -- for example, the number of queries from MySQL over time -- we track the rate between flushed values. While there currently isn't an elegant solution to get raw counts within Datadog, you may want to apply a function to
+your series like cumulative sum or integral. [Read more about Datadog functions][8].
+
+Learn more about the [Count type in the Metrics documentation][4].
+
+### Gauges
+
+Suppose a developer wanted to track the amount of free memory on a machine, we can periodically sample that value as the metric `system.mem.free`:
+
+For Python:
+```python
+
+# Record the amount of free memory every ten seconds.
+while True:
+    statsd.gauge('system.mem.free', get_free_memory())
+    time.sleep(10)
+```
+
+For Ruby:
+```ruby
+# Record the amount of free memory every ten seconds.
+while true do
+    statsd.gauge('system.mem.free', get_free_memory())
+    sleep(10)
+end
+```
+
+Learn more about the [Gauge type in the Metrics documentation][5].
+
+### Histograms
+
+Histograms are specific to DogStatsD. They calculate the statistical distribution of any kind of value, such as the size of files uploaded to your site:
+
+```python
+
+from datadog import statsd
+
+def handle_file(file, file_size):
+  # Handle the file...
+
+  statsd.histogram('mywebsite.user_uploads.file_size', file_size)
+  return
+```
+
+Histograms can also be used with timing data, for example, the duration of a metrics query:
+
+For Python:
+```python
+
+# Track the run time of the database query.
+start_time = time.time()
+results = db.query()
+duration = time.time() - start_time
+statsd.histogram('database.query.time', duration)
+
+# We can also use the `timed` decorator as a short-hand for timing functions.
+@statsd.timed('database.query.time')
+def get_data():
+    return db.query()
+```
+
+For Ruby:
+```ruby
+start_time = Time.now
+results = db.query()
+duration = Time.now - start_time
+statsd.histogram('database.query.time', duration)
+
+# We can also use the `time` helper as a short-hand for timing blocks
+# of code.
+statsd.time('database.query.time') do
+  return db.query()
+end
+```
+
+The above instrumentation produces the following metrics:
+
+- `database.query.time.count`: number of times this metric was sampled
+- `database.query.time.avg`: average time of the sampled values
+- `database.query.time.median`: median sampled value
+- `database.query.time.max`: maximum sampled value
+- `database.query.time.95percentile`: 95th percentile sampled value
+
+These metrics give insight into how different each query time is. We can see how long the query usually takes by graphing the `median`. We can see how long most queries take by graphing the `95percentile`.
```

Comment:
r/`insight`/`insights` ? (i'm not sure)

* Graphing the `median` shows how long the query usually takes. 
* Graphing the `95percentile` shows how long most queries take.